types/budget.d.ts

```diff
@@ -11,6 +11,8 @@ declare global {
      * More info: https://github.com/GoogleChrome/lighthouse/issues/6053#issuecomment-428385930
      */
     export interface Budget {
+      /** Indicates which pages a budget applies to. Uses the robots.txt format */
+      path: string;
```

Comment:
Can we add to the comment of the `Budget` interface that this is the interface post-validation and that the user-facing interface is slightly different? i.e. path is not required there.

Reply:
> should we do a `Budget` (required `path`) and `Budget.Json` (with an optional `path`?). It would be nice to have a type to validate valid input

hmm, actually what if this was`path?: string`? That would keep the internal and external interfaces the same and keep the single-entry budget simple.

- `validatePath()` could accept/return `undefined` or a valid string
- `urlMatchesPattern` could be unchanged except for

```js
  /**
   * Determines whether a URL matches against a robots.txt-style "path".
   * @param {string} url
   * @param {string=} pattern
   * @return {boolean}
   */
  static urlMatchesPattern(url, pattern = '/') {
```